packages/react/src/components/gdpr-consent-dialog/index.tsx

```diff
@@ -0,0 +1,158 @@
+import { GDPRConsent, GDPRConsentLevel } from "@slashid/slashid";
+import { clsx } from "clsx";
+import { useEffect, useReducer } from "react";
+import { useGdprConsent } from "../../hooks/use-gdpr-consent";
+import { publicVariables } from "../../theme/theme.css";
+import { Button } from "../button";
+import { Dialog } from "../dialog";
+import { Cookie } from "../icon/cookie";
+import { Text } from "../text";
+import { Actions } from "./actions";
+import { Settings } from "./settings";
+import {
+  ConsentLevel,
+  GDPR_CONSENT_LEVELS,
+  createInitialState,
+  reducer,
+} from "./state";
+import * as styles from "./style.css";
+
+type Props = {
+  /** Custom class name */
+  className?: string;
+  /** Custom class name for the trigger button */
+  triggerClassName?: string;
+  /** Default open state */
+  defaultOpen?: boolean;
+  /** Value of the lock on the necessary cookies category */
+  necessaryCookiesRequired?: boolean;
+  /** Default consent levels to store when clicking on accept all */
+  defaultAcceptAllLevels?: ConsentLevel[];
+  /** Default consent levels to store when clicking on reject all */
+  defaultRejectAllLevels?: GDPRConsentLevel[];
+  /** Callback when user actions are successful */
+  onSuccess?: (consentLevels: GDPRConsent[]) => void;
+  /** Callback when user actions fail */
+  onError?: (error: unknown) => void;
+  /**
+   * The modality of the dialog.
+   * When set to `true`, interaction with outside elements will be disabled and only dialog content will be visible to screen readers.
+   * When set to `false`, interaction with outside elements will be enabled and all content will be visible to screen readers.
+   */
+  modal?: boolean;
+};
+
+/**
+ * GDPR Consent Dialog component to manage the GDPR consent levels for the current user.
+ */
+export const GDPRConsentDialog = ({
+  className,
+  triggerClassName,
+  onSuccess,
+  onError,
+  defaultOpen = false,
+  necessaryCookiesRequired = false,
+  defaultAcceptAllLevels = [...GDPR_CONSENT_LEVELS],
+  defaultRejectAllLevels = ["none"],
+  modal = true,
+}: Props) => {
+  const { consents, consentState, updateGdprConsent } = useGdprConsent();
+  const [state, dispatch] = useReducer(
+    reducer,
+    createInitialState(consents, defaultOpen)
+  );
+
+  const { consentSettings, open, isLoading, hasError, isCustomizing } = state;
+
+  useEffect(() => {
+    if (!open) {
+      dispatch({ type: "SET_HAS_ERROR", payload: false });
+      dispatch({ type: "SET_IS_CUSTOMIZING", payload: false });
+    }
+  }, [open]);
+
+  useEffect(() => {
+    if (consentState === "ready") {
+      dispatch({
+        type: "SET_CONSENT_SETTINGS",
+        payload: consents,
+      });
+    }
```

Comment:
We should remove `SET_CONSENT_SETTINGS` case in the reducer and use derived state and `useMemo` for `consentSettings` and `open`